crates/swc_ecma_codegen/src/lib.rs

```diff
@@ -703,7 +703,7 @@ where
         self.emit_leading_comments_of_span(num.span(), false)?;
 
         // Handle infinity
-        if num.value.is_infinite() {
+        if num.value.is_infinite() && num.raw.is_none() {
```

Comment:
About the question on Discord, I think some code is removing `Number.raw` .

Reply:
<img width="1840" alt="image" src="https://github.com/user-attachments/assets/dfb03352-2ed9-4032-a0a2-7be1710de2d7">

You can see a list of codes touching `Number.raw` by using vscode feature.

Reply:
As [the playground link](https://play.swc.rs/?version=1.6.7&code=H4sIAAAAAAAAA0vOzysuUahQsFXQNUy1tLS0BgAenTgBEQAAAA%3D%3D&config=H4sIAAAAAAAAA21QOw7CMAzde4rKMwPqwMAdmDiBFVwU1HxkOxJV1buTtA0tEksUv4%2Bf7alpW3iJgWs75W8uIrIQf%2BuMyOgV3xkBHSOJYRsVTpVVKVSPg9ACzSsDivwkLS6S7tx1mwOGEISqY8Oc9bYfj5kmuMgk8issUvTPgf4nMnrpA7tjIyY0egAyZF0MrPeQ2JRG4KO7rrLTLuLk1bqFx6TBoVoDGz039V2iwYVHWmbaTljOtC5%2BgV1Ud%2FzuA1Zu1amcqJk%2FtsEHsI4BAAA%3D) uses `ES2022` as the target and does not enable optimization, you need to check for 

 - swc_ecma_transforms_typescript
 - swc_ecma_transforms_proposal
 - swc_ecma_transforms_base (unlikely, but well)